### PR TITLE
Fix various issues found in font module(2)

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
### Description of the feature or fix

see the commit message

### Checkpoints
- [X] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update CHANGELOG.md
- [ ] Update the documentation

### CODE REVIEW

File: src/lv_font/lv_font_fmt_txt.c [hunk 1]
```diff
@@ -322,8 +317,8 @@ static int32_t kern_pair_8_compare(const void * ref, const void * element)
     const uint8_t * element8_p = element;
 
     /*If the MSB is different it will matter. If not return the diff. of the LSB*/
-    if(ref8_p[0] != element8_p[0]) return (int32_t)ref8_p[0] - element8_p[0];
```
Comment (kisvegabor) [inline src/lv_font/lv_font_fmt_txt.c]: It seemed correct. The values are always big-endian. E.g.
```c
/*Pair left and right glyphs for kerning*/
static const uint16_t kern_pair_glyph_ids[] =
{
    1, 34,
    1, 53,
    1, 58,
    18, 18,
    34, 1,
    34, 53,
    34, 55,
    34, 56,
    34, 58,
    34, 87,
    34, 88,
    34, 90,
    34, 96,
```

Comment (xiaoxiang781216) [reply src/lv_font/lv_font_fmt_txt.c]: Yes, you are right. I misunderstand the spec and comment.

**Comment by xiaoxiang781216** [comment]
@kisvegabor I drop the patch "fix(font): kern_pair_[8|16]_compare should compare the right glyph index first" from PR, please take a look.

**Comment by kisvegabor** [comment]
Great! Thank you!

**Pull request merged**